OpenNMT/onmt/Translator.py

```diff
@@ -48,32 +66,38 @@ def buildTargetTokens(self, pred, src, attn):
 
     def translateBatch(self, batch):
         srcBatch, tgtBatch = batch
-        batchSize = srcBatch.size(0)
+        batchSize = srcBatch.size(1)
         beamSize = self.opt.beam_size
 
         #  (1) run the encoder on the src
 
-        # have to execute the encoder manually to deal with padding
-        encStates = None
-        context = []
-        for srcBatch_t in srcBatch.chunk(srcBatch.size(1), dim=1):
-            encStates, context_t = self.model.encoder(srcBatch_t, hidden=encStates)
-            batchPadIdx = srcBatch_t.data.squeeze(1).eq(onmt.Constants.PAD).nonzero()
-            if batchPadIdx.nelement() > 0:
-                batchPadIdx = batchPadIdx.squeeze(1)
-                encStates[0].data.index_fill_(1, batchPadIdx, 0)
-                encStates[1].data.index_fill_(1, batchPadIdx, 0)
-            context += [context_t]
+        encStates, context = None, None
 
-        encStates = (self.model._fix_enc_hidden(encStates[0]),
-                      self.model._fix_enc_hidden(encStates[1]))
+        if self.model.encoder.num_directions == 2:
+            # bidirectional encoder is negatively impacted by padding
+            # run with batch size 1 for improved translations
+            # This will be resolved when variable length LSTMs are used instead
```

Comment:
Yeah, the brnn encoders will suffer quite a bit unless translate is run with batch size 1 at the moment. I'll add the warning. The unidirectional encoders were always fine.